### PR TITLE
adds task 6

### PR DESCRIPTION
**Pull request by senachary**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -33,4 +33,29 @@ describe("GET api/topics", () => {
                 });
         });
     });
-});
\ No newline at end of file
+});
+
+describe.only("GET /api/articles/:article_id/comments", () => {
```
Comment (2202Hannah) [inline __tests__/app.test.js]: Remove .only from your tests please

File: __tests__/app.test.js [hunk 2]
```diff
@@ -33,4 +33,29 @@ describe("GET api/topics", () => {
...
+                    expect(comments[0]).toHaveProperty("article_id", expect.any(Number))
+                });
+        });
+        test("Should return comments sorted by date in DESC order", () => {
+            return request(app).get("/api/articles/1/comments").expect(200)
+                .then(({ body: { comments } }) => {
+                    expect(comments).toBeSortedBy("created_at", { descending: true })
+            })
+        })
+    })
```
Comment (2202Hannah) [inline __tests__/app.test.js:211]: Please consider the following tests for errors:
> the article_id is invalid e.g. 'not-an-id'
> the article_id is not present in the database e.g. 9999

File: __tests__/app.test.js [hunk 3]
```diff
@@ -33,4 +33,29 @@ describe("GET api/topics", () => {
...
+                    expect(comments[0]).toHaveProperty("author", expect.any(String))
+                    expect(comments[0]).toHaveProperty("body", expect.any(String))
+                    expect(comments[0]).toHaveProperty("article_id", expect.any(Number))
+                });
+        });
+        test("Should return comments sorted by date in DESC order", () => {
+            return request(app).get("/api/articles/1/comments").expect(200)
+                .then(({ body: { comments } }) => {
+                    expect(comments).toBeSortedBy("created_at", { descending: true })
+            })
```
Comment (2202Hannah) [inline __tests__/app.test.js:189]: Please add a test where the article_id exists but there are no comments for that article

**Review by 2202Hannah — CHANGES_REQUESTED** [review]
Hi Sen, there are some additional considerations to be made for this ticket. Please see my in line comments and update your code. Please submit another pull request when you are ready and make sure you have removed .only from your tests.

File: __tests__/app.test.js [hunk 4]
```diff
@@ -33,4 +33,49 @@ describe("GET api/topics", () => {
...
+            return request(app).get("/api/articles/1/comments").expect(200)
+                .then(({ body: { comments } }) => {
+                    expect(comments[0]).toHaveProperty("comment_id", expect.any(Number))
+                    expect(comments[0]).toHaveProperty("votes", expect.any(Number))
+                    expect(comments[0]).toHaveProperty("created_at", "2020-11-03T21:00:00.000Z")
+                    expect(comments[0]).toHaveProperty("author", expect.any(String))
+                    expect(comments[0]).toHaveProperty("body", expect.any(String))
+                    expect(comments[0]).toHaveProperty("article_id", expect.any(Number))
+                });
+        });
```
Comment (B7Bprog) [inline __tests__/app.test.js:184]: We want to make sure that we get all the comments back. Please write a test where you check the length of the array and iterate through it and check each comment for those properties.

File: __tests__/app.test.js [hunk 5]
```diff
@@ -33,4 +33,49 @@ describe("GET api/topics", () => {
...
+        test("Should return comments sorted by date in DESC order", () => {
+            return request(app).get("/api/articles/1/comments").expect(200)
+                .then(({ body: { comments } }) => {
+                    expect(comments).toBeSortedBy("created_at", { descending: true })
+            })
```
Comment (B7Bprog) [inline __tests__/app.test.js:189]: Please add another test where there are no comments for an article, but the request is successful and returning an empty array of comments.

File: controllers/comments-controller.js [hunk 6]
```diff
@@ -0,0 +1,20 @@
+const { selectComments } = require("../models/comments-model")
+
+function getComments(req, res, next) {
+    const article_id = req.params.article_id;
+
+    selectComments(article_id)
+        .then((comments) => {
+            if (comments.length === 0) {
```
Comment (B7Bprog) [inline controllers/comments-controller.js:8]: Error responses should be handled in your error handling middleware in app.js.

File: controllers/comments-controller.js [hunk 7]
```diff
@@ -0,0 +1,20 @@
+const { selectComments } = require("../models/comments-model")
+
+function getComments(req, res, next) {
+    const article_id = req.params.article_id;
```
Comment (B7Bprog) [inline controllers/comments-controller.js:4]: You will need to check the reason for the query returning empty rows:
- article exists but has no comments.
- article does not exist.
You will need to check whether the article exists in order to determine this.
I recommend the use of Promise.all() here for efficiency.

File: __tests__/app.test.js [hunk 8]
```diff
@@ -33,4 +33,49 @@ describe("GET api/topics", () => {
...
+                expect(msg).toBe("Bad Request")
+            })
+        })
+        test("Should return 404: the article_id is not present in the database e.g. 9999", () => {
+            return request(app).get("/api/articles/9999/comments").expect(404)
+                .then(({ body: { msg } }) => {
+                expect(msg).toBe("Not Found")
+            })
+        })
+        test("Should return 404: the article_id exists but there are no comments for that article", () => {
```
Comment (B7Bprog) [inline __tests__/app.test.js]: This should be 200. Please see my comment above recommending this test.

**Review by B7Bprog — CHANGES_REQUESTED** [review]
Hey Sen, please see my comments and refactor your code, then please send us a PR with it.
